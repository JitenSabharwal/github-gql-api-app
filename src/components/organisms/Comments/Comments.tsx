import React from "react"
import styled from "styled-components"
import { Chrono } from "react-chrono"
const Root = styled.div`
  min-width: 45vw;
  height: 700px;
`
type Props = {
  data: {
    createdAt: Date
    body: string
    createdBy: string
  }[]
}
export default function Comments({ data }: Props) {
  return (
    <Root>
      <Chrono
        cardHeight={100}
        items={data.map((d) => {
          return {
            cardTitle: d.createdBy,
            cardSubtitle: d.body,
            title: d.createdAt.toDateString(),
          }
        })}
        mode="VERTICAL"
      />
    </Root>
  )
}
