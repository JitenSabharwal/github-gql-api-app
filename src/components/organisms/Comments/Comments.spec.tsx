import Comments from "./Comments"
import React from "react"
import { shallow } from "enzyme"

describe("Comments", () => {
  let component: React.ReactElement
  beforeAll(() => (component = <Comments />))
  it("Should render Comments", () => {
    const container = shallow(component)
  })
  it("Should render the layout", () => {
    const container = shallow(component)
    expect(container.getElements()).toMatchSnapshot()
  })
})
