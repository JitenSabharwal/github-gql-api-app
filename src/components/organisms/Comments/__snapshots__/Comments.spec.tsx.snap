// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Should render the layout 1`] = `
Array [
  <styled.div />,
]
`;
